From e4ce3fc45013e05d804a7788b7d55b5a56eed3cb Mon Sep 17 00:00:00 2001
From: Bartosz Gotowalski <bartosz.gotowalski@intel.com>
Date: Tue, 14 Oct 2025 17:08:48 +0200
Subject: [PATCH] bump go

---
 examples/go/go.mod | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/examples/go/go.mod b/examples/go/go.mod
index ed43328d7..b8d93c04c 100644
--- a/examples/go/go.mod
+++ b/examples/go/go.mod
@@ -2,4 +2,4 @@ module github.com/protocolbuffers/protobuf/examples/go
 
 go 1.14
 
-require google.golang.org/protobuf v1.27.1
+require google.golang.org/protobuf v1.33.0
-- 
2.43.0

